ext/gmp/gmp.c

```diff
@@ -1350,11 +1350,11 @@ ZEND_FUNCTION(gmp_pow)
 		RETURN_THROWS();
 	}
 
-    double powmax = log((double)ZEND_LONG_MAX);
+	double powmax = 2000;
 
 	if (Z_TYPE_P(base_arg) == IS_LONG && Z_LVAL_P(base_arg) >= 0) {
 		INIT_GMP_RETVAL(gmpnum_result);
-		if ((log(Z_LVAL_P(base_arg)) * exp) > powmax) {
+		if ((log(Z_LVAL_P(base_arg)) / log(256) * exp) > powmax) {
```

Comment:
Don't we just care if it can be "allocated reasonably" ?
Which is actually somewhat described in section 5.14 of the GMP manual (`mpz_export`)
![image](https://github.com/user-attachments/assets/42a36e12-7201-429f-a6bb-2a90b2bd6d17)

Wonder if we shouldn't use a heuristic based on that?

Reply:
This is basically what is already happening in this patch. The difference is that I'm counting bytes, not bits. The logic is to estimate the number of bytes the result would have, and if this is greater than some threshold (`powmax` is badly named; should be rather `max_size_in_bytes` or so), trigger a value error.

`mpz_sizeinbase()` is only used in the else branch, because if the input is a `zend_long`, that wouldn't work. So I'm calculating `log256(base_arg) * exp` instead.

I'll update to counting bytes.